src/app/tests/suites/certification/Test_TC_OO_2_4.yaml

```diff
@@ -238,6 +238,19 @@ tests:
       PICS: OO.S.C00.Rsp
       command: "Off"
 
+      # We get an acknowledgment that the command off above is received, but it takes
+      # a few milliseconds for the off command to actually turn it off. Without this
+      # delay the preceding reboot command might happen before the command is fully
+      # processed resulting in flaky test failure that comes from the readAddribute
+      # verification after the reboot.
+    - label: "Wait for send Off command to take affect"
+      cluster: "DelayCommands"
+      command: "WaitForMs"
+      arguments:
+          values:
+              - name: "ms"
+                value: 10
```

Comment:
And does the test plan need updating for this?

Reply:
I mean a subscription would just be a longer delay really and not actually fix anything. Currently YAML doesn't have the ability to subscribe to something and they wait for a particular value to change before proceeding it just waits for the next read attribute subscription to arrive

Reply:
I mean would you be okay with a readAttribute happening here? I am assuming that the event loop on the server side would have to process the 0ms delay timer before processing the read attribute right?

Reply:
> I mean a subscription would just be a longer delay really and not actually fix anything.

No, it would be a delay until we _know_ the server side has updated the value.

> Currently YAML doesn't have the ability to subscribe to something and they wait for a particular value to change before proceeding it just waits for the next read attribute subscription to arrive

Yes, but if you're only subscribed to the OnOff attribute, then that will be the only thing that arrives, right?  That said, there could be some other bad things that could happen there....

> I am assuming that the event loop on the server side would have to process the 0ms delay timer before processing the read attribute right?

Yes, if we're guaranteed it's a 0ms delay.

Again, any substantive changes to this test, like adding steps, need to go through the CSG and test plan.

Reply:
>Yes, if we're guaranteed it's a 0ms delay.

I think it 0ms delay, I will go about proposing this to the CSG

>Again, any substantive changes to this test, like adding steps, need to go through the CSG and test plan.

Okay

Reply:
Assuming this https://github.com/CHIP-Specifications/chip-test-plans/pull/2464 gets approved I will move to the read Attribute method

Reply:
Chaitanya mentioned that he is okay with this going in with the PICS code `PICS_SDK_CI_ONLY` for now to resolve the test flake and after March 17th, 2023 we can get https://github.com/CHIP-Specifications/chip-test-plans/pull/2464 merged and follow up with changing the PICS code to `OO.S.A0000` after the test plan is updated in the spec